### PR TITLE
Refactor new-keyboard to be python3.7 compatible

### PR DESCRIPTION
**Pull request by Duckle29**
## Description
dirs_exist_ok doesn't exist in python 3.7, which is our minimum required version currently.
copy_tree from distutils has the desired behaviour, and is available in 3.7 however distutils will be deprecated in py3.10 and removed in py3.12

The changes in this PR:
- Walk the directories and apply templates file by file.
- Changed the templates to use printf-style formatting, to be consistent with the rest of QMK
- Changed license header to the smaller SPDX header. (as suggested in https://github.com/qmk/qmk_firmware/pull/14705)
- Added `-n, --realname` if people want their real name in copyright headers

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* https://github.com/qmk/qmk_cli/issues/100

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by Duckle29** [comment]
I'm not sure about using `%(KEYBOARD).c|h` as filenames. From what I've tested it works fine on windows, fedora (wsl2) and qmk wsl

File: data/templates/base/%(KEYBOARD)s.c [hunk 1]
```diff
@@ -0,0 +1,4 @@
+// Copyright %(YEAR)s %(YOUR_NAME)s (@%(USER_NAME)s)
```
Comment (skullydazed) [inline data/templates/base/%(KEYBOARD)s.c:1]: I'm not sure how I feel about % and () in a filename, makes for awkward shell.

File: lib/python/qmk/cli/new/keyboard.py [hunk 2]
```diff
@@ -69,7 +68,7 @@ def new_keyboard(cli):
     # Get username
     user_name = None
     while not user_name:
-        user_name = question('Your Name:', default=find_user_name())
+        user_name = question('Your User Name:', default=find_user_name())
```
Comment (skullydazed) [inline lib/python/qmk/cli/new/keyboard.py:71]:
```suggestion
        user_name = question('Your GitHub User Name:', default=find_user_name())
```

File: data/templates/base/%(KEYBOARD)s.c (hunk 1 above)
Comment (Duckle29) [reply data/templates/base/%(KEYBOARD)s.c:1]: Yeah I agree. Left it in as I figured it'd be discussed then.

**Review by tzarc — APPROVED** [review]

**Pull request merged**